src/index.ts

```diff
@@ -294,6 +295,25 @@ export default class Matic extends SDKClient {
     return this.utils.signEIP712TypedData(typedData, wallet[options.from].privateKey)
   }
 
+  async getSwapSignature(exitNFTId, userAddress, options: SendOptions) {
+    if (!exitNFTId || !userAddress) {
+      throw new Error('NFT id or user address is missing')
+    }
+    if (!options.from) {
+      throw new Error('options.from is missing')
+    }
+
+    const dataHash = this.web3Client.web3.utils.soliditySha3(
+      exitNFTId,
+      this.web3Client.web3.utils.toChecksumAddress(userAddress)
+    )
+
+    const wallet = this.web3Client.getWallet()
+    const vrs = ethUtils.ecsign(ethUtils.toBuffer(dataHash), ethUtils.toBuffer(wallet[options.from].privateKey))
+    let signature = ethUtils.toRpcSig(vrs.v, vrs.r, vrs.s)
+    return signature
```

Comment:
We need to assume that the private key will not be available. In most of the cases, user will give metamask provider and sign things using metamask.